logging: Update to new logging API

ajtowns opened the pull request:
Updates various logging calls to the new API from #28318. All commits are scripted diffs, so should be easy to update if needed, and also easy to reuse the scripts to update other code if needed when rebasing it after this is merged. Changes all uses of `LogPrintLevel()` where the level is hardcoded, and changes all `LogPrintf` and `LogPrint` uses in init.cpp.

ajtowns committed:
scripted-diff: Switch various LogPrintLevel(*,Debug,*) to LogDebug()

-BEGIN VERIFY SCRIPT-
sed -i 's/LogPrintLevel(\(BCLog::[^,]*\), BCLog::Level::Debug,/LogDebug(\1,/' src/net_processing.cpp src/dbwrapper.cpp src/net.cpp src/node/txreconciliation.cpp src/validation.cpp
-END VERIFY SCRIPT-

ajtowns committed:
scripted-diff: Switch various LogPrintLevel(*,Warning,*) to LogWarning()

Changes log output from:

  [net:warning] Something problematic happened

to either

  [warning] Something problematic happened

or, when -loglevelalways=1 is enabled:

  [all:warning] Something problematic happened

-BEGIN VERIFY SCRIPT-
sed -i 's/LogPrintLevel(\(BCLog::[^,]*\), BCLog::Level::Warning, */LogWarning(/' src/net.cpp src/node/blockstorage.cpp src/validation.cpp
-END VERIFY SCRIPT-

ajtowns commented:
Depending on how many conflicts drahtbot collects, maybe this should be targeting a merge around 27.0's feature freeze?

Many of the log statements in init.cpp could be done at a better level, eg `LogPrintf("Error: Out of memory. Terminating.\n");` should obviously be logging at the "error" level, not the "info" level. In order to keep this PR scripted-diff only, I won't fix that here; but it can fixed in an independent PR either before or after this PR is merged.

fanquake commented:
Concept ACK - only 3 conflicts, and none looks like priority projects.

stickies-v committed:
log: use warning level for non-critical log messages

As per doc/developer-notes#logging, LogWarning should be used for
severe problems that do not warrant shutting down the node

stickies-v committed:
log: use error level for critical log messages

As per doc/developer-notes#logging, LogError should be used for
severe problems that require the node to shut down

stickies-v reviewed (approved):
ACK 2632d038754d975e76926b7c068309c7aadc82f5

Since we don't have significant merge conflicts I would this support this PR having a non-scripted diff commit to first fix incorrect levels (instead of having to fix and then re-fix) but am okay with this approach too.

Suggested fixes in https://github.com/bitcoin/bitcoin/compare/master...stickies-v:bitcoin:2024-01/fix-log-level-29231 - happy to open a separate pull for it too?

ajtowns committed:
scripted-diff: Update init.cpp to use LogInfo and LogDebug

-BEGIN VERIFY SCRIPT-
sed -i 's/LogPrint(\(BCLog::[^,]*\),/LogDebug(\1,/' src/init.cpp
sed -i 's/LogPrintf(/LogInfo(/' src/init.cpp
-END VERIFY SCRIPT-

ajtowns committed:
scripted-diff: Switch various LogPrintLevel(*,Error,*) to LogError()

Changes log output from:

  [net:error] Something bad happened

to either

  [error] Something bad happened

or, when -loglevelalways=1 is enabled:

  [all:error] Something bad happened

-BEGIN VERIFY SCRIPT-
sed -i 's/LogPrintLevel(\(BCLog::[^,]*\), BCLog::Level::Error, */LogError(/' src/net.cpp src/txdb.cpp src/txmempool.cpp
-END VERIFY SCRIPT-

ajtowns commented:
> Suggested fixes in [master...stickies-v:bitcoin:2024-01/fix-log-level-29231](https://github.com/bitcoin/bitcoin/compare/master...stickies-v:bitcoin:2024-01/fix-log-level-29231) - happy to open a separate pull for it too?

Added these commits

stickies-v reviewed (approved):
ACK 1441ab74d6cd40e75f1dba80b68bc4b1791bf667

ajtowns commented:
Rebased

epiccurious commented:
> Changes all uses of LogPrintLevel() where the level is hardcoded, and changes all LogPrintf and LogPrint uses in init.cpp.

Trying to understand the concept here. What problem does this solve?

stickies-v commented:
re-ACK 8d46bd6750b47caaf2556f915800c6a22c997bc3 - no change since 1441ab74d6cd40e75f1dba80b68bc4b1791bf667 except for addressing merge conflicts

davidgumberg commented on src/init.cpp:
```diff
@@ -1138,11 +1138,11 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     // Warn about relative -datadir path.
     if (args.IsArgSet("-datadir") && !args.GetPathArg("-datadir").is_absolute()) {
-        LogPrintf("Warning: relative datadir option '%s' specified, which will be interpreted relative to the "
+        LogWarning("Warning: relative datadir option '%s' specified, which will be interpreted relative to the "
```
```suggestion
        LogWarning("Relative datadir option '%s' specified, which will be interpreted relative to the "
```

davidgumberg commented:
utACK https://github.com/bitcoin/bitcoin/commit/8d46bd6750b47caaf2556f915800c6a22c997bc3

Probably tangential to this PR:
 Is `LogPrintLevel` in `void libevent_log_cb(...)` an example of:

> "when [...] a category needs to be added for an info/warning/error log message"

<details>

<summary>Otherwise:</summary>

```diff
 static void libevent_log_cb(int severity, const char *msg)
 {
-    BCLog::Level level;
     switch (severity) {
     case EVENT_LOG_DEBUG:
-        level = BCLog::Level::Debug;
+        LogDebug(BCLog::LIBEVENT, "%s\n", msg);
         break;
     case EVENT_LOG_MSG:
-        level = BCLog::Level::Info;
+        LogInfo("%s\n", msg);
         break;
     case EVENT_LOG_WARN:
-        level = BCLog::Level::Warning;
+        LogWarning("%s\n", msg);
         break;
     default: // EVENT_LOG_ERR and others are mapped to error
-        level = BCLog::Level::Error;
+        LogError("%s\n", msg);
         break;
     }
-    LogPrintLevel(BCLog::LIBEVENT, level, "%s\n", msg);
 }
```

ajtowns commented:
Rebased over #28834, removed redundancy from `LogWarning("Warning: ..")`

laanwj commented:
@ajtowns Sorry, i didn't mean to derail this, i shouldn't have commented.
